Zoo/ZooRelated/AnimalsZone.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Runtime.InteropServices;
+using Zoo.Model.Animals;
+using Zoo.Util.Enum;
+using Zoo.Util.CustomException;
+
+namespace Zoo.ZooRelated
+{
+    internal sealed class AnimalsZone
+    {
+        public string? Name { get; set; }
+
+        public string? Location { get; set; }
+
+        public ZonesType Type { get; }
+
+
+        public List<Animal>? Animals;
+
+
+        public AnimalsZone(List<Animal> animals)
+        {
+            CheckIfAnimalsAreDistinct(animals);
+
+            CheckIfForEachZoneMoreAtLeastThreeAnimals(animals);
+
+            CheckIfAnimalRelatedToThisZone(animals);
+
+            Animals = animals;
+        }
+
+        public AnimalsZone(List<Animal> animals, ZonesType type)
+        {
+            CheckIfAnimalsAreDistinct(animals);
+
+            CheckIfForEachZoneMoreAtLeastThreeAnimals(animals);
+
+            CheckIfAnimalRelatedToThisZone(animals);
+
+            Animals = animals;
+
+            Type = type;
+        }
+
+        private void CheckIfThereAreTwoOrMoreAnimals(List<Animal> animals, Animal Animal)
+        {
+            int CountApperances = CountAnimalTypeInAnimals(animals, Animal.AnimalType);
+            if (CountApperances >= 2)
```

Comment:
Any command using `{}` should have an empty line above and below. For example, `for`, `while`, `if`, etc.